[Refactor] Move Functions from RenderGrid to GridLayoutFunctions 2/n

yulunwu2 opened the pull request:
#### 69e59e394c3ce971a86fbd51ba105a642af3e0e7
<pre>
[Refactor] Move Functions from RenderGrid to GridLayoutFunctions 2/n
<a href="https://bugs.webkit.org/show_bug.cgi?id=297135">https://bugs.webkit.org/show_bug.cgi?id=297135</a>
&lt;<a href="https://rdar.apple.com/157871978">rdar://157871978</a>&gt;

Reviewed by Sammy Gill.

This PR refactors:

availableAlignmentSpaceForGridItemBeforeStretching()

updateAutoMarginsInRowAxisIfNeeded()
updateAutoMarginsInColumnAxisIfNeeded()

out of RenderGrid and into GridLayoutFunctions.

The purpose of this refactor is to clarify interactions between
the  RenderGrid and individual items within RenderBox.

* Source/WebCore/rendering/GridLayoutFunctions.cpp:
(WebCore::GridLayoutFunctions::availableAlignmentSpaceForGridItemBeforeStretching):
(WebCore::GridLayoutFunctions::updateAutoMarginsIfNeeded):
(WebCore::GridLayoutFunctions::updateAutoMarginsInRowAxisIfNeeded):
(WebCore::GridLayoutFunctions::updateAutoMarginsInColumnAxisIfNeeded):
* Source/WebCore/rendering/GridLayoutFunctions.h:
* Source/WebCore/rendering/RenderGrid.cpp:
(WebCore::RenderGrid::layoutGridItems):
(WebCore::RenderGrid::layoutMasonryItems):
(WebCore::RenderGrid::applyStretchAlignmentToGridItemIfNeeded):
(WebCore::RenderGrid::applySubgridStretchAlignmentToGridItemIfNeeded):
(WebCore::RenderGrid::availableAlignmentSpaceForGridItemBeforeStretching const): Deleted.
(WebCore::RenderGrid::updateAutoMarginsInRowAxisIfNeeded): Deleted.
(WebCore::RenderGrid::updateAutoMarginsInColumnAxisIfNeeded): Deleted.
* Source/WebCore/rendering/RenderGrid.h:

Canonical link: <a href="https://commits.webkit.org/298545@main">https://commits.webkit.org/298545@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a3da2f9915d889bf90f232bb0e0cb2a96179c9ab

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/115797 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/35452 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/25989 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/121845 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/66324 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/593a71b9-5659-4c7d-8ce7-e8ec319a5534) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/117686 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/36139 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/44038 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/87968 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/42602 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/f2923820-a31d-4e8a-bad9-99413b5d42af) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/118745 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/28836 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/103916 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/68380 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/27975 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/22029 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/65516 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/98228 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/22152 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/124997 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/42684 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/32030 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/96721 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/43051 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/100098 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/96506 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24559 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/41764 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/19633 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/38611 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/42574 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/48169 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/42045 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/45379 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/43752 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/917f3b7fbcd8a4f9498386f32da33b69de7d619c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49174

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/742bc7c2ed4be158c5b4fdae99ab4a1c5dfb8b59. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49174

sammygill commented on Source/WebCore/rendering/RenderGrid.cpp:
```diff
@@ -1571,8 +1571,8 @@ void RenderGrid::layoutGridItems(GridLayoutState& gridLayoutState)
         // We need pending layouts to be done in order to compute auto-margins properly.
-        updateAutoMarginsInColumnAxisIfNeeded(gridItem);
-        updateAutoMarginsInRowAxisIfNeeded(gridItem);
+        GridLayoutFunctions::updateAutoMarginsInColumnAxisIfNeeded(gridItem, writingMode());
+        GridLayoutFunctions::updateAutoMarginsInRowAxisIfNeeded(gridItem, writingMode());
```
Might be good to store the writing mode in a local variable so we could avoid calling the function twice

sammygill commented on Source/WebCore/rendering/RenderGrid.cpp:
```diff
@@ -1616,8 +1616,8 @@ void RenderGrid::layoutMasonryItems(GridLayoutState& gridLayoutState)
         // We need pending layouts to be done in order to compute auto-margins properly.
-        updateAutoMarginsInColumnAxisIfNeeded(gridItem);
-        updateAutoMarginsInRowAxisIfNeeded(gridItem);
+        GridLayoutFunctions::updateAutoMarginsInColumnAxisIfNeeded(gridItem, writingMode());
+        GridLayoutFunctions::updateAutoMarginsInRowAxisIfNeeded(gridItem, writingMode());
```
Might be good to store the writing mode in a local variable so we could avoid calling the function twice

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/a3da2f9915d889bf90f232bb0e0cb2a96179c9ab. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/49174

yulunwu2 committed:
[Refactor] Move Functions from RenderGrid to GridLayoutFunctions 2/n
https://bugs.webkit.org/show_bug.cgi?id=297135
<rdar://157871978>

Reviewed by Sammy Gill.

This PR refactors:

availableAlignmentSpaceForGridItemBeforeStretching()

updateAutoMarginsInRowAxisIfNeeded()
updateAutoMarginsInColumnAxisIfNeeded()

out of RenderGrid and into GridLayoutFunctions.

The purpose of this refactor is to clarify interactions between
the  RenderGrid and individual items within RenderBox.

* Source/WebCore/rendering/GridLayoutFunctions.cpp:
(WebCore::GridLayoutFunctions::availableAlignmentSpaceForGridItemBeforeStretching):
(WebCore::GridLayoutFunctions::updateAutoMarginsIfNeeded):
(WebCore::GridLayoutFunctions::updateAutoMarginsInRowAxisIfNeeded):
(WebCore::GridLayoutFunctions::updateAutoMarginsInColumnAxisIfNeeded):
* Source/WebCore/rendering/GridLayoutFunctions.h:
* Source/WebCore/rendering/RenderGrid.cpp:
(WebCore::RenderGrid::layoutGridItems):
(WebCore::RenderGrid::layoutMasonryItems):
(WebCore::RenderGrid::applyStretchAlignmentToGridItemIfNeeded):
(WebCore::RenderGrid::applySubgridStretchAlignmentToGridItemIfNeeded):
(WebCore::RenderGrid::availableAlignmentSpaceForGridItemBeforeStretching const): Deleted.
(WebCore::RenderGrid::updateAutoMarginsInRowAxisIfNeeded): Deleted.
(WebCore::RenderGrid::updateAutoMarginsInColumnAxisIfNeeded): Deleted.
* Source/WebCore/rendering/RenderGrid.h:

Canonical link: https://commits.webkit.org/298545@main
